translator/util/ec2util/ec2util.go

```diff
@@ -115,5 +121,18 @@ func (e *ec2Util) deriveEC2MetadataFromIMDS() error {
 		fmt.Println("E! [EC2] Fetch identity document from EC2 metadata fail:", err)
 	}
 
+	// Instance tags: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html#instance-metadata-ex-7
+	if tagKeys, err := md.GetMetadata("tags/instance"); err == nil {
+		for _, tagKey := range strings.Fields(tagKeys) {
+			if tagValue, err := md.GetMetadata("tags/instance/" + tagKey); err == nil {
+				e.Tags[tagKey] = tagValue
+			} else {
+				fmt.Printf("E! [EC2] Fetch instance tag %s from EC2 metadata fail:%v\n", tagKey, err)
+			}
+		}
+	} else {
+		fmt.Println("E! [EC2] Fetch instance tags from EC2 metadata fail:", err)
```

Comment:
Just to be clear, my understanding is we don't want to return an error here because retrieving instance tags from IMDS requires customers to explicitly opt-in, right? Or was `return err` missed here?

Reply:
Yes - it's a soft error based on the opt-in. I felt like this should not be a hard error because the pattern in the file seems to be just printing the errors.